src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1515,9 +1515,18 @@ private File createSshKeyFile(File key, SSHUserPrivateKey sshUser) throws IOExce
     }
 
     private String quoteWindowsCredentials(String str) {
-        // Assumes the only meaningful character is %, this may be
-        // insufficient.
-        return str.replace("%", "%%");
+        // Quote special characters for Windows Batch Files
```

Comment:
I'm absolutely willing to make methods package protected for tests.  The CliGitAPIImpl class is generally not accessed directly, rather it is accessed through either the GitClient interface or the GitAPI interface, so it is unlikely that someone will mistakenly place their new class into the same class as CliGitAPIImpl, and then use that package protected method.

I started an [evaluation branch](https://github.com/MarkEWaite/git-client-plugin/commits/master-PR231-escape-windows-chars) with some preliminary tests.  They are incomplete, and riddled with problems.  You're welcome to use those for ideas, or to discard them and create the tests that you think are most sensible.  I won't be able to return to that code for at least a week.